frontend/src/components/common/Resource/EditorDialog.tsx

```diff
@@ -557,3 +587,263 @@ export default function EditorDialog(props: EditorDialogProps) {
 export function ViewDialog(props: Omit<EditorDialogProps, 'onSave'>) {
   return <EditorDialog {...props} onSave={null} />;
 }
+
+interface UploadYamlProps {
+  setUploadYaml: (value: boolean) => void;
+  setCode: React.Dispatch<React.SetStateAction<{ code: string; format: string }>>;
+}
+
+function UploadYourYaml(props: UploadYamlProps) {
+  const { setUploadYaml, setCode } = props;
+  const { t } = useTranslation();
+  const [uploadedfile, setUploadedFile] = React.useState<File | null>(null);
+
+  const Uploadyamlfromfilesystem = () => {
```

Comment:
this should be in camel case
```suggestion
  const uploadYamlFromFilesystem = () => {
```

Reply:
It's a component, so it should start with a capital letter.